src/main/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperations.scala

```diff
@@ -292,4 +294,62 @@ object TezosDatabaseOperations extends LazyLogging {
   def selectDistinctLike(table: String, column: String, matchingString: String)(implicit ec: ExecutionContext): DBIO[List[String]] = {
     sql"""SELECT DISTINCT #$column FROM #$table WHERE #$column LIKE '%#$matchingString%'""".as[String].map(_.toList)
   }
+
+
+  /** Implicit value that allows getting table row as Map[String, Any] */
+  private implicit val getMap: GetResult[Map[String, Any]] = GetResult[Map[String, Any]](positionedResult => {
+    val metadata = positionedResult.rs.getMetaData
+    (1 to positionedResult.numColumns).flatMap(i => {
+      val columnName = metadata.getColumnName(i).toLowerCase
+      val columnValue = positionedResult.nextObjectOption
+      columnValue.map(columnName -> _)
+    }).toMap
+  })
+
+  /**
+    * Selects elements filtered by the predicates
+    * THIS METHOD IS VULNERABLE TO SQL INJECTION
+    * @param table          name of the table
+    * @param columns        list of column names
+    * @param predicates     list of predicates for query to be filtered with
+    * @return               list of map of [string, any], which represents list of rows as a map of column name to value
+    */
+  def selectWithPredicates(table: String, columns: List[String], predicates: List[Predicates])(implicit ec: ExecutionContext):
+  DBIO[List[Map[String, Any]]] = {
+    val pred = predicates.map { p =>
+      concat(
+        sql""" AND #${p.field} #${QueryProtocolTypes.mapOperationToSQL(p.operation, p.inverse)} """,
+        List(values(p.set.map(_.toString))))
+    }
+    val query = sql"""SELECT #${columns.mkString(",")} FROM #$table WHERE true """
+
+    concat(query, pred).as[Map[String, Any]].map(_.toList)
+  }
+
+  //some adjusted hacks from https://github.com/slick/slick/issues/1161 as Slick does not have simple concatenation of actions
+  /** concatenates SQLActionsBuilders */
+  @tailrec
+  private def concat(acc: SQLActionBuilder, actions: List[SQLActionBuilder]): SQLActionBuilder = {
```

Comment:
Maybe this whole set of support functions might be put into a "Util" class that could make all this more readable... slick is awful at composing interpolated queries, it seems

Reply:
moved to `DatabaseUtil`